pkg/sql/crdb_internal.go

```diff
@@ -903,6 +903,22 @@ WITH
 	INNER JOIN latestpayload AS payload ON j.id = payload.job_id
 	LEFT JOIN latestprogress AS progress ON j.id = progress.job_id
 `
+
+	// systemJobsAndJobInfoBaseQueryWithIDPredicate is the same as
+	// systemJobsAndJobInfoBaseQuery but with a predicate on `job_id` in the CTE
+	// queries.
+	systemJobsAndJobInfoBaseQueryWithIDPredicate = `
+WITH
+	latestpayload AS (SELECT job_id, value FROM system.job_info AS payload WHERE info_key = 'legacy_payload'::BYTES AND job_id = $1),
```

Comment:
Why don't these two get the same order/limit clauses?

Reply:
rah SQL is hard. I should give them the order clause. I didn't give them the limit clause because the driver uses a row iterator so it _can_ handle more than one result. The caller of `update.go` used `QueryRow` and so it would panic if it saw more than one result. 

That said I now realize this means 4 rows in `SHOW JOBS` for the same job_id which is no good. Is it weird that `SELECT * from crdb_internal.system_jobs WHERE id = $1` doesn't show all the rows corresponding to id $1?

Reply:
@dt this was a good call out, even when we do not have a predicate we need to ensure we only display a single row per job. To make this happen I've added a `distinct` clause to the `id` column along with the `ORDER BY` clauses to the CTEs. I also added some more checks to the test.